indico/modules/events/registration/client/js/form/fields/DateInput.jsx

```diff
@@ -0,0 +1,131 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2021 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import TimePicker from 'rc-time-picker';
+import React, {useState} from 'react';
+
+import {SingleDatePicker} from 'indico/react/components';
+import {FinalDropdown} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+
+export default function DateInput({htmlName, id, disabled, dateFormat, hasTime}) {
+  const [date, setDate] = useState(null);
+  const [time, setTime] = useState(null);
+  const uses24HourFormat = false;
```

Comment:
Should we make this a new field setting so the event organizer can decide what they prefer? Since we do not use a locale-specific format, I think 12/24h format should be up to the organizer as well.

This could be simply a checkbox "Use 24h time format". It should be disabled when the selected choice in the dropdown does not include a time component.